### DISCUSSION TITLE
Goertzel filter (or other) to detect specific frequency?

### DISCUSSION THREAD

**Discussion by SoundNerdsUnite** [pschatzmann/arduino-audio-tools#1783] [Q&A] [answered]
Hi everyone,

I'm workng on a sketch that would use the Goertzel algorithm to detect a specific frequency, and then do something (serial print, turn on LED, etc..). In hopes to contribute to this wonderful library.

Has anyone tried this before?
Does the library already have such a feature, maybe using FFT, etc..

Thanks!

**Comment by pschatzmann** [comment, marked as answer]
There is a FFT Wiki: I suggest to try this first because this is the most powerful functionality.
You can also try out the [FrequncyAutoCorrelationStream](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_frequncy_auto_correlation_stream.html) class

**Reply by SoundNerdsUnite** [reply]
Great! Thank you, Phil!
